src/chapters/orm-relationships/many-to-many.rst

```diff
@@ -0,0 +1,690 @@
+
+.. admonition:: Note
+
+   If you want to verify what code this video starts with, check out the `event-detail-view <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-detail-view>`_ branch. If you want to verify what code this video ends with, check out the `event-tag-model <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-tag-model>`_ branch.
+
+   The starter code for this video is found at the `event-detail-view branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-detail-view>`_ of ``CodingEventsDemo``. The final code presented in this video is found on the `event-tag-model branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/event-tag-model>`_. As always, code along to the videos on your own ``CodingEvents`` project.
+
+.. todo: Add event tag video
+
+The ``EventTag`` Model - Text
+-----------------------------
+
+.. index:: ! join class
+
+To model a join table for ``Event`` and ``Tag`` classes, we will create a **join class**. Given our discussion of join tables above, we know that it will need ``EventId`` and ``TagId`` properties. To more easily work with the corresponding ``Event`` and ``Tag`` objects in our controllers, we will also include properties of those specific types.
+
+.. sourcecode:: csharp
+   :linenos:
+
+   using System;
+   namespace CodingEventsDemo.Models
+   {
+      public class EventTag
+      {
+
+         public int EventId { get; set; }
+         public Event Event { get; set; }
+
+         public int TagId { get; set; }
+         public Tag Tag { get; set; }
+
+         public EventTag()
+         {
+         }
+      }
+   }
+
+To make this class persistent, and a new ``DbSet`` entry to ``EventDbContext``:
+
+.. sourcecode:: csharp
+   :lineno-start: 11
+
+   public DbSet<EventTag> EventTags { get; set; }
+
+.. index:: composite primary key
+
+Since our join table will make use of a composite primary key, we need to add some additional configuration to ``EventDbContext``.
+
+.. sourcecode:: csharp
+   :lineno-start: 18
+
+   protected override void OnModelCreating(ModelBuilder modelBuilder)
+   {
+      modelBuilder.Entity<EventTag>()
+            .HasKey(et => new { et.EventId, et.TagId });
+   }
+
+The method ``OnModelCreating`` can be overridden from the base class, ``DbContext``, in order to provide additional configuration for the data store. In this case, we add code that configures ``EntityTag`` to have a composite primary key consisting of the properties/columns ``EventId`` and ``TagId``.
+
+This completes configuration of our join class. Before proceeding, create and apply a database migration. After doing so, verify that there is a new join table, ``EventTag``.
+
+Adding a ``Tag`` to an ``Event`` - Video
+----------------------------------------
+
+Now that we have established a many-to-many relationship between ``Event`` and ``Tag`` we can write controller and view code to allow users to add tags to events.
```

Comment:
Now that we have established a many-to-many relationship between ``Event`` and ``Tag``,